test/e2e/reloadinterval_test.go

```diff
@@ -0,0 +1,173 @@
+
+		// set ReloadInterval to 0, which should give default of 5s
+		{"set reloadInterval default", metav1.Duration{0 * time.Second}, "5s"},
+
+		// set ReloadInterval to a value outside of allowed bounds
+		{"set reloadInterval 130s", metav1.Duration{130 * time.Second}, "2m"},
+		// {"set reloadInterval 0.5s", metav1.Duration{500 * time.Millisecond}, "1s"},
+		// {"set reloadInterval 1s", metav1.Duration{-1 * time.Second}, "1s"},
+	} {
+		t.Log(testCase.description)
+		if err := setReloadIntervalValidValue(t, kclient, 1*time.Minute, testCase.reloadInterval, icName); err != nil {
+			t.Fatalf("failed to update ingresscontroller with reloadInterval=%v: %v", testCase.reloadInterval, err)
+		}
+		if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, availableConditionsForPrivateIngressController...); err != nil {
+			t.Fatalf("failed to observe expected conditions: %v", err)
+		}
+		if err := waitForDeploymentEnvVar(t, kclient, deployment, 1*time.Minute, ingresscontroller.RouterReloadIntervalEnvName, testCase.expectedEnvVar); err != nil {
+			t.Fatalf("router deployment not updated with %s=%v: %v", ingresscontroller.RouterReloadIntervalEnvName, testCase.expectedEnvVar, err)
+		}
+	}
+}
+
+// Tests values that do not pass
+// the regex validation (i.e. values that result in an error).
+func TestReloadIntervalInvalidValues(t *testing.T) {
+	t.Parallel()
+
+	const expectedErr string = `spec.tuningOptions.reloadInterval in body should match '^(0|([0-9]+(\.[0-9]+)?(s|m))+)$'`
+
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "reloadinterval-invalid-vals"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newPrivateController(icName, domain)
+
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller %s: %v", icName, err)
+	}
+
+	defer assertIngressControllerDeleted(t, kclient, ic)
+	t.Logf("waiting for ingresscontroller %v", icName)
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, availableConditionsForPrivateIngressController...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	ic, err := getIngressController(t, kclient, icName, 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get ingress controller: %v", err)
+	}
+
+	deployment := &appsv1.Deployment{}
+	if err := kclient.Get(context.TODO(), controller.RouterDeploymentName(ic), deployment); err != nil {
+		t.Fatalf("failed to get ingresscontroller router deployment: %v", err)
+	}
+
+	for _, testCase := range []struct {
+		description    string
+		reloadInterval metav1.Duration
+		expectedEnvVar string
+	}{
+		{"set reloadInterval 500 ms", metav1.Duration{500 * time.Millisecond}, "5s"},
+		{"set reloadInterval -1s", metav1.Duration{-1 * time.Second}, "5s"},
+	} {
+		t.Log(testCase.description)
+		err := setReloadIntervalInvalidValue(t, kclient, testCase.reloadInterval, icName)
```

Comment:
Also, rather than duplicate this, you should add it to the function I described in the previous comment, passing in 5, kclient, icName, deployment, and testcases as input.